testing/test_recwarn.py

```diff
@@ -284,3 +284,27 @@ def test(run):
         ''')
         result = testdir.runpytest()
         result.stdout.fnmatch_lines(['*2 passed in*'])
+
+    def test_message_using(self):
+        source = "warnings.warn('my runtime warning', RuntimeWarning)"
+        pytest.warns(RuntimeWarning, source, message='my runtime warning')
```

Comment:
warns should not replicate the old call style , and even more so for the execute code snippets style, its just an accident of history we should stay clear of